src/components/PayOnchain.tsx

```diff
@@ -1,73 +1,121 @@
 import { BigNumber } from "bignumber.js";
-import { Show } from "solid-js";
+import { Show, createMemo, createResource } from "solid-js";
 
 import CopyButton from "../components/CopyButton";
 import QrCode from "../components/QrCode";
 import { BTC } from "../consts/Assets";
-import { Denomination } from "../consts/Enums";
+import { SwapType } from "../consts/Enums";
 import { useGlobalContext } from "../context/Global";
-import { formatAmount } from "../utils/denomination";
-import { clipboard, cropString, isMobile } from "../utils/helper";
+import { getPairs } from "../utils/boltzClient";
+import { formatAmount, formatDenomination } from "../utils/denomination";
+import { clipboard, cropString, getPair, isMobile } from "../utils/helper";
+import LoadingSpinner from "./LoadingSpinner";
 
 const PayOnchain = (props: {
-    asset: string;
+    type: SwapType;
+    assetSend: string;
+    assetReceive: string;
     expectedAmount: number;
     address: string;
     bip21: string;
 }) => {
-    const { t, denomination, separator } = useGlobalContext();
+    const { t, denomination, separator, setPairs, pairs } = useGlobalContext();
+
+    const [pairsFetch] = createResource(async () => {
+        if (pairs() !== undefined) {
```

Comment:
Why not use `fetchPairs` from the global context here?
https://github.com/BoltzExchange/boltz-web-app/blob/8f6e924dc88a23d0fb33b016978acda5d6daf5bf/src/context/Global.tsx#L78